linetools/isgm/abscomponent.py

```diff
@@ -100,6 +118,48 @@ def from_abslines(cls, abslines, stars=None, comment="", reliability='none', **k
         if len(abslines) > 1:
             for absline in abslines[1:]:
                 slf.add_absline(absline, **kwargs)
+
+        ### Add attribs to component
+        # First check that the measurements for all the lines match
+        # Grab them all
+        cols = np.array([al.attrib['N'].value for al in abslines])
+        colerrs = np.array([al.attrib['sig_N'].value for al in abslines])
+        bs = np.array([al.attrib['b'].value for al in abslines])
+        berrs = np.array([al.attrib['sig_b'].value for al in abslines])
+        vels = np.array([al.attrib['vel'].value for al in abslines])
+        velerrs = np.array([al.attrib['sig_vel'].value for al in abslines])
+        medcol = np.median(cols)
+        medcolerr = np.median(colerrs)
+        medb = np.median(bs)
+        medberr = np.median(berrs)
+        medvel = np.median(vels)
+        medvelerr = np.median(velerrs)
+        # Perform checks on measurements
+        colcrit = all(np.abs(cols - medcol) / medcol < tol) == True
+        bcrit = all(np.abs(bs - medb) / medb < tol) == True
+        # Set attribs
+        slf.attrib['N'] = medcol / u.cm ** 2
+        slf.attrib['sig_N'] = medcolerr / u.cm ** 2
+        slf.attrib['b'] = medb * u.km / u.s
+        slf.attrib['sig_b'] = medberr * u.km / u.s
+        slf.attrib['vel'] = medvel * u.km / u.s
+        slf.attrib['sig_vel'] = medvelerr * u.km / u.s
+        logN, sig_logN = ltaa.log_clm(slf.attrib)
+        slf.attrib['logN'] = logN
+        slf.attrib['sig_logN'] = sig_logN
+        # Stop or throw warnings
+        if (bcrit&colcrit):
+            pass
+        elif chk_meas:
+            raise ValueError('The line measurements for the lines in this '
+                             'component are not consistent with one another.')
+            print(slf)
+        elif medb != 0: # Empty b-values (upper lims) can throw off bcrit
```

Comment:
Are you sure medb should be zero?

Reply:
Yeah, it's so that the code doesn't barf when upper limits go through the routine.